internal/docstore/doc.go

```diff
@@ -48,4 +48,44 @@
 // If a struct doesn't have a DocstoreRevision field, then the logic described above
 // won't apply to documents read and written with that struct. All writes with the
 // struct will succeed even if the document was changed since the last Get.
+//
+//
+// Queries
+//
+// Docstore supports querying within a collection. Call the Query method on
+// Collection to obtain a Query value, Then build your query by calling Query methods
+// like Where, Limit and so on. Finally, call the Get method on the query to execute it.
+// The result in an iterator, whose use is described below.
+//
+//     iter := coll.Query().Where("size", ">", 10).Limit(5).Get(ctx)
+//
+// The Where methods defines a filter condition, much like a WHERE clause in SQL. Conditions
+// are of the form "field op value", where field is any document field path (including dot-separated
+// paths), op is one of "=", ">", "<", ">=" or "<=", and value can be a string or number.
+//
+// Although you can make multiple Where calls, not all provider systems can handle them.
+// For some providers, Where clauses may be processed on the client.
+//
+// The Limit method specifies the maximum number of results to return.
+//
+// Docstore provides no guarantees about the ordering of results. There is no way to
+// ask for a particular ordering.
+//
+// Use the DocumentIterator returned from Query.Get by repeatedly calling its Next
+// method until it returns io.EOF. Always call Stop when you are finished with an
```

Comment:
Do you always have to call `Stop`, or only if you stopped before `io.EOF`? IIRC the `Stop` docstring said the latter. I don't feel strongly either way, but let's make sure we're consistent. 

Also, if it's "always call Stop", consider adding a `SetFinalizer` for it to log if it is GCd without `Stop` being called, similar to `Message` without `Ack` and `blob.Reader/Writer` without `Close`.

Reply:
If the iterator returns EOF, can the driver clean up the resource if needed?

Reply:
We could make it unnecessary for EOF, but let's say "always." That's easier to remember, matches file Close, and leads to more maintainable code (if you defer the Stop, and then a maintainer changes the iteration to end early, the code still works).

Reply:
I updated the `DocumentIterator` documentation to match.